### PR TITLE
feat(net): allow regexp to whitelist interfaces

### LINKED ISSUE #39: feature request: ignore virtual interface
Hey

I know I can ignore interfaces like this

```yaml
net-opts:
  interfaces:
    - eth0
```

But predicting interface names can be a bit difficult. I'ld like to filter out the many many interfaces created by docker

image

Can you add an option to filter out virtual interface ?

Or maybe allowing regexpr ?

### PR DESCRIPTION
**Pull request by aurrelhebert**
fix #39

### CODE REVIEW

File: collectors/net.go [hunk 1]
```diff
@@ -168,6 +170,16 @@ func stringInSlice(str string, list []string) bool {
 		if v == str {
 			return true
 		}
+
+		if strings.HasPrefix(strings.TrimSpace(v), "~") {
+
+			matched, _ := regexp.MatchString(strings.Replace(strings.TrimSpace(v), "~", "", 1), str)
+
```
Comment (miton18) [inline collectors/net.go]: `return matched`

Comment (aurrelhebert) [reply collectors/net.go]: Nope :) return matched will not process other elements of to compare list. You can only return matched in case of success ;)

**Comment by rgarrigue** [comment]
@aurrelhebert in which version will it be released ?

**Comment by aurrelhebert** [comment]
Hi @rgarrigue, it was released in the last Noderig version (2.5.1): https://github.com/ovh/noderig/releases/tag/2.5.1

**Comment by rgarrigue** [comment]
Ok, but when will it hits http://last.public.ovh.metrics.snap.mirrors.ovh.net/debian/pool/main/n/noderig/ ?

**Comment by aurrelhebert** [comment]
Hello @rgarrigue we were having some issues with our automated deployment of Noderid. Noderig is now packaged and released on our mirror. 
Thanks for the wait

**Pull request merged**